ckanext/dcatapchharvest/profiles.py

```diff
@@ -768,16 +789,41 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
             #  Simple values
             items = [
                 ('status', ADMS.status, None, Literal),
-                ('rights', DCT.rights, None, Literal),
                 ('coverage', DCT.coverage, None, Literal),
-                ('license', DCT.license, None, Literal),
                 ('identifier', DCT.identifier, None, Literal),
                 ('media_type', DCAT.mediaType, None, Literal),
                 ('spatial', DCT.spatial, None, Literal),
             ]
 
-            self._add_triples_from_dict(resource_dict, distribution, items)
+            if resource_dict.get('rights'):
+                rights_uri = dh.get_license_uri_by_name(
+                    resource_dict.get('rights')
+                )
+                if rights_uri is not None:
+                    rights_ref = URIRef(rights_uri)
+                    g.add((rights_ref, RDF.type, DCT.RightsStatement))
+                    g.add((distribution, DCT.rights, rights_ref))
+                if rights_uri is None:
+                    resource_rights_ref = URIRef(resource_dict.get('rights'))
+                    g.add((resource_rights_ref, RDF.type, DCT.RightsStatement))
+                    g.add((distribution, DCT.rights, resource_rights_ref))
+
+            if resource_dict.get('license'):
+                license_uri = dh.get_license_uri_by_name(
+                    resource_dict.get('license')
+                )
+                if license_uri is not None:
+                    license_ref = URIRef(license_uri)
+                    g.add((license_ref, RDF.type, DCT.LicenseDocument))
+                    g.add((distribution, DCT.license, license_ref))
+                if license_uri is None:
+                    resource_license_ref = URIRef(resource_dict.get('license'))
```

Comment:
Same here, we need to check if the license belongs to the vocabulary before we can map it.

Reply:
added the check with the following commit. You are right.